From 62f7822834818a391cabcf7fa276f45a75d37065 Mon Sep 17 00:00:00 2001
From: balajhidn <balajhix.neelakantan.durvas@intel.com>
Date: Mon, 17 Oct 2022 12:43:23 +0000
Subject: [PATCH] Add redfish events for BMC secure FW update

Added new redfish events
 - If any partition is marked as bad.
 - Secure boot status if it is enabled or disabled.
 - If a key retired. key index N got retired.
 - SVN updated to version N(new SVN version).

Tested:
Redfish events are logged successfully and verified
the same in Redfish interface (Postman tool)

Signed-off-by: Gade-Rajasekhar Reddy <raja.sekhar.reddy.gade@intel.com>
Signed-off-by: Aqeeb Khan <aqeebx.khan@intel.com>
Upstream-Status: Pending
---
 .../registries/openbmc_message_registry.hpp   | 45 +++++++++++++++++++
 1 file changed, 45 insertions(+)

diff --git a/redfish-core/include/registries/openbmc_message_registry.hpp b/redfish-core/include/registries/openbmc_message_registry.hpp
index daeb8c66..1c766e46 100644
--- a/redfish-core/include/registries/openbmc_message_registry.hpp
+++ b/redfish-core/include/registries/openbmc_message_registry.hpp
@@ -656,6 +656,51 @@ constexpr std::array registry = {
                      {"string", "string"},
                      "None.",
                  }},
+    MessageEntry{"BMCPartitionCorrupted",
+        {
+            "Indicates bmc partition corrupted.",
+            "%1 partition corrupted.",
+            "Critical",
+            1,
+            {"string"},
+            "None.",
+        }},
+    MessageEntry{"BMCSecureBootEnabled",
+        {
+            "Indicates BMC secure boot is enabled.",
+            "BMC secure boot is enabled.",
+            "OK",
+            0,
+            {},
+            "None.",
+        }},
+    MessageEntry{"BMCSecureBootDisabled",
+        {
+            "Indicates BMC secure boot is disabled.",
+            "BMC secure boot is disabled.",
+            "Critical",
+            0,
+            {},
+            "None.",
+        }},
+    MessageEntry{"SecureBootKeyRetired",
+        {
+            "Indicates the Secure Boot Key got retired",
+            "Key with index %1 got retired successfully",
+            "OK",
+            1,
+            {"string"},
+            "None.",
+        }},
+    MessageEntry{"SVNProgrammedToOTP",
+        {
+            "Indicates Security Version Number programmed to OTP memory",
+            "Programming SVN to OTP memory is successful. Updated SVN in OTP memory is %1",
+            "OK",
+            1,
+            {"string"},
+            "None.",
+        }},
     MessageEntry{
         "GeneralFirmwareSecurityViolation",
         {
-- 
2.25.1

